### ISSUE TITLE
403 on loading gist

### ISSUE THREAD

**Issue by daveliepmann** [mhuebert/maria#204]
### Issue:
>[this FP gist from Josh Glover](https://www.maria.cloud/gist/975be504eed35898e2dfa85e5b8b56be) doesn't seem to load for me; Console shows a 403 error?

source: [tweet](https://twitter.com/lemonodor/status/1088193257723449345)

### Environment:

  - Browser:
  - Platform (Mac/Windows/Linux):
  

### URL to a gist replicating the issue (if possible/relevant)  
https://www.maria.cloud/gist/975be504eed35898e2dfa85e5b8b56be

**Comment by daveliepmann** [comment]
I can't replicate myself in macOS 10.14.2 Mojave under Chrome, Firefox, or Safari, but I believe we've seen failure-to-load before under certain browser settings or similar.

**Comment by mhuebert** [comment]
This can be caused by rate limits in the GitHub API, which we use to load gists:

https://developer.github.com/v3/#rate-limiting

> For API requests using Basic Authentication or OAuth, you can make up to 5000 requests per hour. Authenticated requests are associated with the authenticated user, regardless of whether Basic Authentication or an OAuth token was used. This means that all OAuth applications authorized by a user share the same quota of 5000 requests per hour when they authenticate with different tokens owned by the same user.
>
> For unauthenticated requests, the rate limit allows for up to 60 requests per hour. Unauthenticated requests are associated with the originating IP address, and not the user making requests.

If this is the case, there should be a ‘X-RateLimit-Remaining: 0’ header in the error response. If not, then we’ll have to keep looking. 

Traffic is currently low enough that we could also consider proxying requests ourselves, vs. having users talk to GitHub directly.

**Comment by wiseman** [comment]
When I saw the error I was using Chrome 71.0.3578.98 (Official Build) (64-bit) on macOS 10.14.2, with uBlock Origin 1.17.4.  I was logged into GitHub. The page just displayed something along the lines of "Loading gist...".  When I checked the developer console, there was a log message describing an HTTP 403 error (sorry I didn't save it!).

I haven't been able to reproduce since the initial experience.
